packages/overlays/src/OverlayController.js

```diff
@@ -1050,6 +1050,31 @@ export class OverlayController extends EventTargetShim {
     if (this.__hasActiveTrapsKeyboardFocus) {
       return;
     }
+
+    // Reset contain focus util when contentNode content has changed
+    // and tab detection element is missing
+    if (!this.trapsKeyboardObserver) {
```

Comment:
Could this logic be moved into `src/utils/contain-focus.js`, so that it's completely self contained as a util?

Reply:
Updated

Reply:
Nice, test as well if I would be nitpicking. But I approved :)

Reply:
I added one test in OverlayController test to kinda end to end test it by doing
```
ctrl.contentNode.innerHTML = `my content`;
```
and then verifying that the tab detection el is still there in the contentNode (reinserted)

Or do you mean a different test?

Reply:
There was a test purely for the util if I'm correct

Reply:
https://github.com/ing-bank/lion/blob/master/packages/overlays/test/utils-tests/contain-focus.test.js

Reply:
Ahh hm okay I moved the test to contain-focus.test.js since we should indeed unit test the contain-focus part and not only check if it works in conjunction with OverlayController